### DISCUSSION TITLE
C++ concepts ERROR

### DISCUSSION THREAD

**Discussion by rockwotj** [tree-sitter/tree-sitter#2863] [Q&A]
Looking at the tree sitter playground here: https://tree-sitter.github.io/tree-sitter/playground

The follow C++ snippet results in ERRORs:

```cpp
template<typename T> requires Addable<T> && Subtractable<T> // conjunctions
T f(T);
```

Which is strange because it looks like there are tests for this in tree-sitter-cpp:

https://github.com/tree-sitter/tree-sitter-cpp/blob/a71474021410973b29bfe99440d57bcd750246b1/test/corpus/concepts.txt#L99-L100

What is going on? I'm running into this because I'm getting errors in ast-grep for these.

**Comment by alaviss** [comment]
As far as I can tell they have not updated the parsers in the playground for awhile.
